### DISCUSSION TITLE
Skip second _find_key for ReadWrite

### DISCUSSION THREAD

**Discussion by bpmckinnon** [greg7mdp/parallel-hashmap#222] [Ideas]
I'm not sure about the internals for how memory is flagged as available, but right now when using the ReadWrite lock if the switch_to_unique returns true, then _find_key is run again.  Would it make sense to 
`store the capacity (or pointer) of the set`
`do _find_key`
`if offset is valid` 
`    switch the lock to writer`
`    if the capacity (or pointer) has changed or the entry at offset no longer matches`
`        repeat _find_key`
Then you wouldn't have to repeat the find unless your data has been invalidated.

**Comment by greg7mdp** [comment]
That's an interesting idea, but I'm not sure it would be much faster. I think the overhead for doing the find again is quite low.
